python/pyiceberg/catalog/__init__.py

```diff
@@ -25,6 +26,41 @@
 from pyiceberg.table.partitioning import UNPARTITIONED_PARTITION_SPEC, PartitionSpec
 from pyiceberg.table.sorting import UNSORTED_SORT_ORDER, SortOrder
 from pyiceberg.typedef import EMPTY_DICT, Identifier, Properties
+from pyiceberg.utils.config import Config, merge_config
+
+logger = logging.getLogger(__name__)
+
+_env_config = Config()
+
+
+def load_catalog(name: str, uri: str, **properties: str | None) -> Catalog:
+    from pyiceberg.catalog.rest import RestCatalog
+
+    supported_catalogs: dict[str, type[Catalog]] = {"http": RestCatalog}
+
+    try:
+        # In case Thrift isn't installed
+        from pyiceberg.catalog.hive import HiveCatalog
+
+        supported_catalogs["thrift"] = HiveCatalog
+    except ImportError:
+        logger.warning("Apache Hive not supported, to enable: pip install 'pyiceberg[hive]'")
+
+    if name:
```

Comment:
I think name should always be required. If we are creating some default catalog, we can use a default name (or randomly generate one).

Reply:
I already changed this back, but this check was still in place. I think mypy should also have caught this since the else branch is unreachable.